### PR TITLE
Update rules_rust to v0.47.1 (with Rust v1.79.0).

### PR DESCRIPTION
**Pull request by PiotrSikora**

### CODE REVIEW

File: bazel/cargo/remote/BUILD.ahash-0.8.3.bazel [hunk 1]
```diff
@@ -11,15 +11,15 @@ load("@rules_rust//rust:defs.bzl", "rust_library")
...
-# licenses([
-#     "TODO",  # MIT OR Apache-2.0
-# ])
-
 rust_library(
     name = "ahash",
-    srcs = glob(["**/*.rs"]),
+    srcs = glob(
+        include = ["**/*.rs"],
+        allow_empty = True,
```
Comment (mpwarres) [inline bazel/cargo/remote/BUILD.ahash-0.8.3.bazel:18]: Mostly just curious: why is `allow_empty = True` now needed in numerous places? I thought that is the default setting for the glob() function.

File: bazel/repositories.bzl [hunk 2]
```diff
@@ -16,11 +16,18 @@ load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
 load("@bazel_tools//tools/build_defs/repo:utils.bzl", "maybe")
 
 def proxy_wasm_rust_sdk_repositories():
+    maybe(
+        http_archive,
+        name = "rules_cc",
```
Comment (mpwarres) [inline bazel/repositories.bzl:21]: Why is this needed? I didn't see use of a cc build rule added anywhere.

File: bazel/cargo/remote/BUILD.ahash-0.8.3.bazel (hunk 1 above)
Comment (PiotrSikora) [reply bazel/cargo/remote/BUILD.ahash-0.8.3.bazel:18]: No idea, this is auto-generated by `//bazel/cargo:crates_vendor`.

File: bazel/repositories.bzl (hunk 2 above)
Comment (PiotrSikora) [reply bazel/repositories.bzl:21]: It's not added anywhere in this PR, but there is an existing reference in the auto-generated `bazel/cargo/remote/alias_rules.bzl` to `@rules_cc`.

I think `rules_rust` got more picky about it, because it fails without adding it, but I didn't investigate further:
```
$ bazelisk --noworkspace_rc run --noenable_bzlmod //bazel/cargo:crates_vendor
```
```
ERROR: Failed to load Starlark extension '@@rules_cc//cc:action_names.bzl'.
Cycle in the workspace file detected. This indicates that a repository is used prior to being defined.
The following chain of repository dependencies lead to the missing definition.
 - @@rules_cc
This could either mean you have to add the '@@rules_cc' repository with a statement like `http_archive` in your WORKSPACE file (note that transitive dependencies are not added automatically), or move an existing definition earlier in your WORKSPACE file.
ERROR: Error computing the main repository mapping: cycles detected during computation of main repo mapping
```

**Comment by PiotrSikora** [comment]
(force pushed to update email address, no code changes)

File: bazel/cargo/remote/BUILD.ahash-0.8.3.bazel (hunk 1 above)
Comment (mpwarres) [reply bazel/cargo/remote/BUILD.ahash-0.8.3.bazel:18]: Got it, thanks.

File: bazel/repositories.bzl (hunk 2 above)
Comment (mpwarres) [reply bazel/repositories.bzl:21]: I see, SGTM.

**Review by mpwarres — APPROVED** [review]
